### PR TITLE
MAINT-40063 : fix news formatMention when no text after @ char

### PR DESCRIPTION
**Pull request by hakermi**
This PR should fix **ArrayIndexOutOfBoundsException** raised by the formatMention method when it tries to split text contains @ char and extract the username of the split table.

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 1]
```diff
@@ -1204,11 +1204,15 @@ public String formatMention(String text) {
...
-        if (identity != null) {
-          next = next.split("@")[0] + "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
-                  + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
+        String username = "";
+        String[]splitTable = next.split("@");
+        if(splitTable.length>1) {
+          username = splitTable[1];
+          org.exoplatform.social.core.identity.model.Identity identity = loadUser(username);
+          if (identity != null) {
+            next = next.split("@")[0] + "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
        String[] splitTable = next.split("@");
        if (splitTable.length > 1) {
          username = splitTable[1];
          org.exoplatform.social.core.identity.model.Identity identity = loadUser(username);
          if (identity != null) {
            next = splitTable[0] + "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
```

**Review by ahamdi — CHANGES_REQUESTED** [review]
Please add :
 - description to the PR
 - A Unit Test to cover the fix and prove it does no more happen

**Review by plamarque — COMMENTED** [review]
unit tests covering the cases handled by the change seem missing.

**Comment by hakermi** [comment]
> Please add :
> 
> * description to the PR
> * A Unit Test to cover the fix and prove it does no more happen

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 2]
```diff
@@ -1205,11 +1205,15 @@ public String formatMention(String text) {
...
       } else if (next.contains("@")) {
-        String username = next.split("@")[1];
-        org.exoplatform.social.core.identity.model.Identity identity = loadUser(username);
-        if (identity != null) {
-          next = next.split("@")[0] + "<a href=\"" + CommonsUtils.getCurrentDomain() + identity.getProfile().getUrl() + "\">"
-                  + encoder.encodeHTML(identity.getProfile().getFullName()) + "</a>";
+        String username = "";
+        String[]splitTable = next.split("@");
+        if(splitTable.length>1) {
+          username = splitTable[1];
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]:
```suggestion
        String[] splitTable = next.split("@");
        if (splitTable.length > 1) {
          username = splitTable[1];
```

**Review by ahamdi — DISMISSED** [review]

**Review by ahamdi — APPROVED** [review]

**Pull request merged**